core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -262,7 +262,9 @@ final class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: Stri
       0
     else
       watchers.tryCompleteWatched()
-    debug(s"Request key $key unblocked $numCompleted $purgatoryName operations")
+    if (numCompleted > 0) {
+      debug(s"Request key $key unblocked $numCompleted $purgatoryName operations")
```

Comment:
Before this change, we'll see many useless logs like this:
` DEBUG Request key GroupJoinKey(test.group) unblocked 0 Rebalance operations`
We should log when `numCompleted > 0`.